src/azure-cli/azure/cli/command_modules/aro/tests/latest/test_aro_commands.py

```diff
@@ -95,130 +29,311 @@ def test_aro_show(self, resource_group):
             'name': name,
+            'vnet': self.create_random_name('vnet', 10),
             'resource_group': resource_group,
             'subscription': subscription,
             'master_subnet': master_subnet,
             'worker_subnet': worker_subnet,
             'master_ip_range': '10.{}.{}.0/24'.format(randint(0, 127), randint(0, 255)),
             'worker_ip_range': '10.{}.{}.0/24'.format(randint(0, 127), randint(0, 255)),
-            'master_subnet_resource': resource_id(subscription=subscription, resource_group=resource_group, namespace='Microsoft.Network', type='virtualNetworks', child_type_1='subnets', name='dev-vnet', child_name_1=master_subnet),
-            'worker_subnet_resource': resource_id(subscription=subscription, resource_group=resource_group, namespace='Microsoft.Network', type='virtualNetworks', child_type_1='subnets', name='dev-vnet', child_name_1=worker_subnet),
         })
 
-        self.cmd('network vnet create -g {rg} -n dev-vnet --address-prefixes 10.0.0.0/9')
-        self.cmd('network vnet subnet create -g {rg} --vnet-name dev-vnet -n {master_subnet} --address-prefixes {master_ip_range} --service-endpoints Microsoft.ContainerRegistry')
-        self.cmd('network vnet subnet create -g {rg} --vnet-name dev-vnet -n {worker_subnet} --address-prefixes {worker_ip_range} --service-endpoints Microsoft.ContainerRegistry')
-        self.cmd('network vnet subnet update -g {rg} --vnet-name dev-vnet -n {master_subnet} --disable-private-link-service-network-policies true')
-
-        self.cmd('aro create -g {rg} -n {name} --master-subnet {master_subnet_resource} --worker-subnet {worker_subnet_resource} --subscription {subscription} --tags test=show')
 
-        self.cmd('aro show -g {rg} -n {name} --subscription {subscription}', checks=[
-            self.check('name', '{name}'),
-            self.check_pattern('id', '.*{name}'),
-            self.check('resourceGroup', '{rg}')
+        self.cmd('network vnet create -g {rg} -n {vnet} --address-prefixes 10.0.0.0/9')
+        master_subnet_id = self.cmd('network vnet subnet create -g {rg} --vnet-name {vnet} -n {master_subnet} --address-prefixes {master_ip_range} --service-endpoints Microsoft.ContainerRegistry').get_output_in_json().get('id')
+        worker_subnet_id = self.cmd('network vnet subnet create -g {rg} --vnet-name {vnet} -n {worker_subnet} --address-prefixes {worker_ip_range} --service-endpoints Microsoft.ContainerRegistry').get_output_in_json().get('id')
+        self.cmd('network vnet subnet update -g {rg} --vnet-name {vnet} -n {master_subnet} --disable-private-link-service-network-policies true')
+
+        self.kwargs.update(
+            {
+                'master_subnet_resource': master_subnet_id,
+                'worker_subnet_resource': worker_subnet_id,
+            }
+        )
+
+        # test aro create
+        with mock.patch('azure.cli.command_modules.aro._rbac._gen_uuid', side_effect=self.create_guid):        
```

Comment:
Why do we have so many changes? I was expecting only `aro create` be wrapped with a mock. Could you please explain rest of the changes?

Reply:
The big amount of changes goes on my head. I decided to group changes with the second issue the ARO test has. 

Having the `aro create` in every test is tedious and requires quite a lot of computing power and money.

Therefore I have collected all tests in the single test function and sorted them in order so they can be safely tested.

Thus only one `aro create` is required.

Reply:
The comment leftovers are going to be removed as soon as I see the CI results.